src/app/public/testing/modal-fixture.spec.ts

```diff
@@ -0,0 +1,494 @@
+  </sky-modal-footer>
+</sky-modal>
+
+<!-- This span is important to test the querySelectorAll function -->
+<span
+  class="test-class"
+  id="test-modal-content-2"
+>
+  Non-modal content
+</span>
+`
+})
+class TestModalComponent {}
+
+@Component({
+  selector: 'modal-test',
+  template: `
+<button
+  class="sky-btn sky-btn-primary"
+  id="test-launch-button"
+  type="button"
+  (click)="launchModal()"
+>
+  Launch modal
+</button>
+`
+})
+class TestComponent {
+
+  public ariaDescribedBy: string;
+
+  public ariaLabelledBy: string;
+
+  public ariaRole: string;
+
+  public fullPage: boolean;
+
+  public helpKey: string;
+
+  public size: string;
+
+  public tiledBody: boolean;
+
+  private modalInstance: SkyModalInstance;
+
+  constructor(private modalService: SkyModalService) {}
+
+  public launchModal() {
+    this.modalInstance = this.modalService.open(TestModalComponent, {
+      ariaDescribedBy: this.ariaDescribedBy,
+      ariaLabelledBy: this.ariaLabelledBy,
+      ariaRole: this.ariaRole,
+      helpKey: this.helpKey,
+      size: this.size,
+      fullPage: this.fullPage,
+      tiledBody: this.tiledBody
+    });
+
+    this.modalInstance.helpOpened.subscribe((key: string) => {
+      this.helpTriggered(key);
+    });
+  }
+
+  public closeModal() {
+    this.modalInstance.close();
+  }
+
+  public helpTriggered(key: string) {
+    return;
+  }
+
+}
+
+@NgModule({
+  declarations: [
+    TestComponent,
+    TestModalComponent
+  ],
+  imports: [
+    RouterTestingModule,
+    SkyModalModule
+  ],
+  providers: [
+    {
+      provide: SkyThemeService,
+      useClass: ModalMockThemeService
+    }
+  ],
+  entryComponents: [
+    TestModalComponent
+  ]
+})
+class TestModule { }
+
+//#endregion Test component
+
+describe('Modal fixture', () => {
+
+  let fixture: ComponentFixture<TestComponent>;
+  let modalService: SkyModalService;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        TestModule
+      ]
+    });
+
+    modalService = TestBed.inject(SkyModalService);
+  });
+
+  // This is necessary as due to modals being launched outside of the test bed they will not
+  // automatically be disposed between tests.
+  afterEach(() => {
+    fixture.componentInstance.closeModal();
+    fixture.detectChanges();
+
+    // NOTE: This is important as it ensures that the modal host component is fully disposed of
+    // between tests. This is important as the modal host might need a different set of component
+    // injectors than the previous test.
+    modalService.dispose();
+  });
+
+  it('should retun the `ariaDescribedBy` property correctly', fakeAsync(() => {
+    fixture = TestBed.createComponent(
+      TestComponent
+    );
+
+    fixture.detectChanges();
+
+    fixture.componentInstance.ariaDescribedBy = 'describingID';
+    fixture.componentInstance.launchModal();
+
+    fixture.detectChanges();
+    tick();
+    fixture.detectChanges();
+
+    const modal = new SkyModalFixture(
```

Comment:
Could we componentize this to reduce some redundancy? Something like.. `const modal = launchModal()`? And the helper function would look something like...
```
launchModal(): SkyModalFixture {
    fixture.componentInstance.launchModal();

    fixture.detectChanges();
    tick();
    fixture.detectChanges();

    const modal = new SkyModalFixture(
      fixture,
      'test-modal'
    );

    fixture.detectChanges();

    return modal;
}
```

Reply:
Done